sycl/include/sycl/ext/oneapi/properties/property_utils.hpp

```diff
@@ -106,93 +109,17 @@ template <typename RHS> struct SelectNonVoid<void, RHS> {
   using type = RHS;
 };
 
-// Merges two tuples by recursively extracting the type with the minimum
-// PropertyID in the two tuples and prepending it to the merging of the
-// remaining elements.
-template <typename T1, typename T2> struct Merge {};
-template <typename... LTs> struct Merge<std::tuple<LTs...>, std::tuple<>> {
-  using type = std::tuple<LTs...>;
+// Sort types accoring to their PropertyID.
+struct SortByPropertyId {
+  template<typename T, typename U> using fn =
```

Comment:
Can we use the same T1/T2 as on the left? I'm still not sure what our policies regarding those identifiers are.